gherkin/java/src/main/java/io/cucumber/gherkin/PickleCompiler.java

```diff
@@ -181,7 +203,17 @@ private PickleDocString pickleDocString(DocString docString, List<TableCell> var
         );
     }
 
-    private PickleStep pickleStep(Step step, List<TableCell> variableCells, TableRow valuesRow) {
+
+    private static final Map<StepKeywordType, PickleStepType> pickleStepTypeFromKeywordType = new HashMap<>();
+
+    static {
```

Comment:
Rather than here, look at how ew handle two other 'similar' enums that we need to cast between. It makes sense to me to handle the same way 

https://github.com/cucumber/common/blob/main/gherkin/java/src/main/java/io/cucumber/gherkin/Parser.java#L75

Reply:
I verified that works. However, it hard-codes the ordering assumption between `keywordType` and `type` enum values, without that being explicit in the code base elsewhere. The way it's coded now, that ordering isn't required, reducing the likelihood of breaking the code base by changing the enum order in the schema.

Reply:
Concluding for now that I'd rather keep it as it is for now, also because the cast is part of the enum declaration, which for `PickleStepType` is the result of code generation -- in contrast to the code you reference above.